tests/learning/test_file_instruction_finetuning_data_repository.py

```diff
@@ -0,0 +1,120 @@
+import random
+
+from intelligence_layer.core import Language, Message
+from intelligence_layer.learning import (
+    FileInstructionFinetuningDataRepository,
+    InstructionFinetuningSample,
+    InstructionFinetuningSample_,
+    InstructionFinetuningSampleAttributes,
+    RawInstructionFinetuningSample,
+)
+
+
+def test_file_instruction_finetuning_data_repository_can_store_load_and_delete_sample(
+    file_instruction_finetuning_data_repository: FileInstructionFinetuningDataRepository,
+    instruction_finetuning_sample: InstructionFinetuningSample,
+) -> None:
+    file_instruction_finetuning_data_repository.store_sample(
+        instruction_finetuning_sample
+    )
+    loaded_sample = file_instruction_finetuning_data_repository.sample(
+        instruction_finetuning_sample.id
+    )
+
+    assert instruction_finetuning_sample == loaded_sample
+
+    file_instruction_finetuning_data_repository.delete_sample(
+        instruction_finetuning_sample.id
+    )
+    no_sample_expected = file_instruction_finetuning_data_repository.sample(
+        instruction_finetuning_sample.id
+    )
+
+    assert no_sample_expected is None
+
+
+def test_file_instruction_finetuning_data_repository_can_store_load_and_delete_samples(
+    file_instruction_finetuning_data_repository: FileInstructionFinetuningDataRepository,
+    raw_instruction_finetuning_sample: RawInstructionFinetuningSample,
+) -> None:
+    samples = [
+        InstructionFinetuningSample.from_raw_sample(raw_instruction_finetuning_sample)
+        for _ in range(10)
+    ]
+    ids = [sample.id for sample in samples]
+
+    file_instruction_finetuning_data_repository.store_samples(samples)
+    loaded_samples = file_instruction_finetuning_data_repository.samples(ids)
+
+    assert set(ids) == set(loaded_sample.id for loaded_sample in loaded_samples)
+
+    file_instruction_finetuning_data_repository.delete_samples(ids)
+    no_samples_expected = file_instruction_finetuning_data_repository.samples(ids)
+
+    assert list(no_samples_expected) == []
+
+
+def test_file_instruction_finetuning_data_repository_can_show_first_n_samples(
+    file_instruction_finetuning_data_repository: FileInstructionFinetuningDataRepository,
+    raw_instruction_finetuning_sample: RawInstructionFinetuningSample,
+) -> None:
+    n = 5
+    samples = [
+        InstructionFinetuningSample.from_raw_sample(raw_instruction_finetuning_sample)
+        for _ in range(n)
+    ]
+
+    file_instruction_finetuning_data_repository.store_samples(samples)
+    head = list(file_instruction_finetuning_data_repository.head(n))
+
+    assert len(head) == n
```

Comment:
After increasing the number of the stored samples I would also like not only to check for the correct length of the retrieved samples but also that they are the correct ones

Reply:
Not possible (in a straight forward way) due to the parallelization of tests. Many samples are inserted at the same time so there isn't a way of asserting which samples are actually returned.

Reply:
Ah, I see. Then it's fine by me.